src/model.js

```diff
@@ -2138,6 +2138,7 @@ Specify a different name for either index to resolve this issue.`);
     }
 
     const countOptions = Utils.cloneDeep(options);
+    this._injectScope(countOptions);
```

Comment:
If I remove this line and run the new test on sqlite it still passes. So I'm not sure if this fixes the issue or that the test is incorrect. Can you take a look at it?

Reply:
Oh lemme look into that, I do think that the tests are wrong since. Lemme see if it just that or am I messing something up.

Reply:
Hey @WikiRik seems like the tests are fine but I think what I lack in them is to produce the wanted case, tbh I am not able to come up with an example of the case. Here's why!

So the issue was that the query that was being generated was ambiguous, the example is the following:

```js
Player.scope('includeTeam').findAndCountAll({ distinct: true });
```

The query generated for the above is the following:

```sql
SELECT count(DISTINCT(`id`)) AS `count` FROM `Players` AS `Player` LEFT OUTER JOIN `Teams` AS `Team` ON `Player`.`TeamTeamId` = `Team`.`team_id`;
```

Notice above that the query contains ``DISTINCT(`id`)``, here `id` is ambiguous in the sense that if I were NOT to assign explicit primary key, the autogenerated field `id` can be of either of `Team` or `Player` table.

However post the update the query generated is the following!

```sql
SELECT count(DISTINCT(`Player`.`id`)) AS `count` FROM `Players` AS `Player` LEFT OUTER JOIN `Teams` AS `Team` ON `Player`.`TeamId` = `Team`.`id`;
```

![Screenshot 2022-11-11 at 2 59 44 PM](https://user-images.githubusercontent.com/25434993/201310154-d295ce94-c13c-47a6-b485-799110eb3acc.png)

This kinda says that the tests are kinda redundant but the query generated is specific. So I am kinda stuck with creating the exact case. Any feedback would be good!

@ephys sorry for the ping but what's your take on the above!?